routing/control_tower.go

```diff
@@ -102,8 +107,9 @@ func (s *ControlTowerSubscriber) Close() {
 type controlTower struct {
 	db *channeldb.PaymentControl
 
-	subscribers    map[lntypes.Hash][]*ControlTowerSubscriber
-	subscribersMtx sync.Mutex
+	subscribersAllPayments []*ControlTowerSubscriber
```

Comment:
So I went to check the unique ID mentioned in the review club. Turned out other implementations involve a db write, such as `htlcswitch/sequencer`, which could be an overkill here. So I think we could either implement a monotonically increasing integer here as our unique ID, or use `uuid`(which seems to be much simpler) if we decide to replace the slice with a map here.

Reply:
Thanks for checking! I noticed uuid is not used anywhere else in the codebase. So maybe the monotonically increasing integer would be the best way to go. If you'd not reboot LND for one year, you could do 584942417 unique updates per millisecond. Which is a ridiculous number, so a uint64 should be safe to use here.

Reply:
>  If you'd not reboot LND for one year, you could do 584942417 unique updates per millisecond.

Interesting. How did you come up with this figure?

> Which is a ridiculous number, so a uint64 should be safe to use here.

I don't think the number of the updates matter here since the ID is used to track the number of payments subscriptions? Ideally a single all-payments subscriber should be enough to fit all.

Reply:
> Interesting. How did you come up with this figure?
I was just checking how many unique numbers you can create with this and how many subscriptions you could do in the time LND is running. I was thinking if you have some weird subscription client that keeps subscribing/unsubscribing, how long will it take until the number rolls over. The answer is: ridiculously long, basically. That's where that came from. It's ulong64 maxvalue / number of milliseconds in a year.

> I don't think the number of the updates matter here
So yeah that was about how many updates to the subscriptions were made.